src/nanovg_gl.h

```diff
@@ -363,36 +366,52 @@ static GLNVGtexture* glnvg__allocTexture(GLNVGcontext* gl)
 			gl->textures = textures;
 			gl->ctextures = ctextures;
 		}
-		tex = &gl->textures[gl->ntextures++];
+		idx = gl->ntextures++;
 	}
 
+	tex = &gl->textures[idx];
 	memset(tex, 0, sizeof(*tex));
-	tex->id = ++gl->textureId;
+	// pre-increment so salt begins at one. When GLNVGcontext is
+	// allocated initially it is set to 0 via memset.
+	tex->salt = ++gl->textureSalt;
```

Comment:
The reason for using a salt per texture entry is that they are less likely to wrap around and keep unique. I'd prefer to use that version.